pkg/controller/synchronize.go

```diff
@@ -40,9 +42,19 @@ func Synchronize(opts *SynchronizeOpts) error {
 	}
 
 	desiredTree.ApplyFunction(applyDeclaration(opts.ClusterDeclaration), &resourcetree.ResourceNode{})
+
+	if opts.DeleteAll {
```

Comment:
I'd suggest creating a `DeleteTree` function as an alternative to `Synchronize`. A delete tree function would be alot simpler and we wouldn't pollute the synchronize with lots of repetative if's

Reply:
I would like to hear more! I am totally up for a rewrite of this stuff